tests/countries/test_nepal.py

```diff
@@ -0,0 +1,522 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.constants import PUBLIC, WORKDAY
+from holidays.countries.nepal import Nepal, NP, NPL
+from tests.common import CommonCountryTests
+
+
+class TestNepal(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1924, 2050)
+        super().setUpClass(Nepal, years=years)
+        cls.workday_holidays = Nepal(categories=WORKDAY, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(Nepal, NP, NPL)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Nepal(years=1923, categories=(PUBLIC, WORKDAY)))
+
+    def test_special_holidays(self):
+        name_day_of_national_mourning = "Day of National Mourning"
+
+        for dt, name in (
+            ("1999-02-09", name_day_of_national_mourning),
+            ("2011-03-06", name_day_of_national_mourning),
+            ("2016-02-10", name_day_of_national_mourning),
+            ("2023-01-16", name_day_of_national_mourning),
+            ("2023-02-13", "People War's Day"),
+            ("2023-09-14", name_day_of_national_mourning),
+        ):
+            self.assertHolidayName(name, dt)
+
+    def test_prithvi_jayanti(self):
+        name = "Prithvi Jayanti"
+
+        # Public Holidays.
+        self.assertHolidayName(name, (f"{year}-01-11" for year in range(2021, 2050)))
+        self.assertNoHolidayName(name, (f"{year}-01-11" for year in range(2019, 2021)))
+
+        # Workdays.
+        self.assertHolidayName(
+            name, self.workday_holidays, (f"{year}-01-11" for year in range(2019, 2021))
+        )
+        self.assertNoHolidayName(
+            name, self.workday_holidays, (f"{year}-01-11" for year in range(2021, 2050))
+        )
```

Comment:
```suggestion
        self.assertHolidayName(
            name, self.workday_holidays, (f"{year}-01-11" for year in range(2019, 2021))
        )
        self.assertNoHolidayName(name, self.workday_holidays, range(1924, 2019), range(2021, 2050))
```
And the same for other similar holidays.